ibis/backends/bigquery/__init__.py

```diff
@@ -350,7 +350,20 @@ def exists_table(self, name: str, database: str = None) -> bool:
 
     def fetch_from_cursor(self, cursor, schema):
         query = cursor.query
-        df = query.to_arrow().to_pandas(timestamp_as_object=True)
+        query_result = query.result()
```

Comment:
@tswast Can you take a look at this block of code and say whether this is expected behavior?

The use case is reading from in `bigquery-public-data.hacker_news.comments`, but having `ibis-gbq` be the billing project.

Without this workaround, the storage API creates a read session in the data project (`bigquery-public-data`), which causes queries to fail when using the pyarrow functionality.

Reply:
I wouldn't expect this to be necessary. If the query succeeded, then I assume the billing project is being set correctly in the client constructor https://github.com/ibis-project/ibis/blob/01bd4026e463984ab9fd17ea6d57ee6e42d50b8b/ibis/backends/bigquery/__init__.py#L169 and in the query method https://github.com/ibis-project/ibis/blob/01bd4026e463984ab9fd17ea6d57ee6e42d50b8b/ibis/backends/bigquery/__init__.py#L236

I've filed https://github.com/googleapis/python-bigquery/issues/1422 to investigate this further, but I think it's fine to keep this workaround if there really is a bug.

Reply:
Update: I think there really is a bug. The project from "client" is used instead of the project from the QueryJob.

Reply:
Ah, okay. Thanks Tim. I'll keep this comment unresolved so the link is easier to find.